sankaku/clients/http_client.py

```diff
@@ -1,23 +1,59 @@
+import os
+from typing import Dict
+
 from aiohttp import ClientSession
+from aiohttp_retry import ExponentialRetry, RetryClient
 from loguru import logger
 
-from .abc import ABCHttpClient
 from sankaku import errors, constants as const
+from sankaku.constants import BASE_RETRIES
 from sankaku.models.http import ClientResponse
-
+from .abc import ABCHttpClient
 
 __all__ = ["HttpClient"]
 
+try:
+    from aiohttp_socks import ProxyConnector as SocksProxyConnector
+except (ImportError, ModuleNotFoundError):
+    SocksProxyConnector = None
+
+
+def _get_socks_connector():
+    if SocksProxyConnector is not None:
+        proxy = os.environ.get('ALL_PROXY') or os.environ.get('HTTPS_PROXY') or os.environ.get('HTTP_PROXY')
+        if proxy.startswith('socks'):
+            return SocksProxyConnector.from_url(proxy)
+        else:
+            return None
+    else:
+        return None
+
 
 class HttpClient(ABCHttpClient):
     """HTTP client for API requests that instances use a single session."""
+
     def __init__(self) -> None:
-        self.headers: dict[str, str] = const.HEADERS.copy()
-        self.session: ClientSession = ClientSession()
+        self.headers: Dict[str, str] = const.HEADERS.copy()
+
+        socks_connector = _get_socks_connector()
+        if socks_connector:
```

Comment:
Add explicit comparison of `socks_connector`: `if socks_connector is not None`. The reason is [PEP8](https://peps.python.org/pep-0008/#programming-recommendations) recommendation:

> Also, beware of writing if x when you really mean if x is not None – e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

Reply:
Changed